acme/dns_challenge_rfc2136.go

```diff
@@ -99,3 +101,55 @@ func (r *DNSProviderRFC2136) changeRecord(action, fqdn, value string, ttl int) e
 
 	return nil
 }
+
+// findZone determines the zone of a qualifying cert DNSname
+func (r *DNSProviderRFC2136) findZone(fqdn string) (string, error) {
+	// Do we have it cached?
+	if val, ok := r.domain2zone[fqdn]; ok {
+		return val, nil
+	}
+
+	// Query the authorative nameserver for a hopefully non-existing SOA record,
+	// in the authority section of the reply it will have the SOA of the
+	// containing zone. rfc2308 has this to say on the subject:
+	//   Name servers authoritative for a zone MUST include the SOA record of
+	//   the zone in the authority section of the response when reporting an
+	//   NXDOMAIN or indicating that no data (NODATA) of the requested type exists
+	m := new(dns.Msg)
+	m.SetQuestion(fqdn, dns.TypeSOA)
+	m.SetEdns0(4096, false)
+	m.RecursionDesired = true
```

Comment:
Right, found it. Not happy. I have a split view bind setup, an internal one with, well all the internal hosts and an external view with all the other domains I have, exposed to the outside world. Turns out that when I query the NS from inside my network, it correctly picks the internal view which knows nothing about the external domains. So without recursion I get the root NSs as hint to look at.

As adding those zones into the internal view  is a no-no due to the dynamic updates we want to do later on, I'm opting for leaving the recursion on.